plugins/dependencies/src/test/kotlin/de/fayard/refreshVersions/rules/BundledDependenciesTest.kt

```diff
@@ -1,59 +1,73 @@
+        val validatedDependencyMapping = validatedDependencyMappingFile.readLines()
+            .map { it.substringBeforeLast(':') }
+            .filter { it.isNotBlank() }
+            .toSet()
 
-        https://github.com/jmfayard/refreshVersions/runs/872495471?check_suite_focus=true
-        https://gradle.com/s/l47xwdefpipo2
-        */
+        // "standard repositories" are mavenCentral and google
+        val reposUrls = listOf(
+            "https://repo.maven.apache.org/maven2/",
+            "https://dl.google.com/dl/android/maven2/",
+            "https://plugins.gradle.org/m2/"
+            //"https://jcenter.bintray.com/",
+        )
 
-        runBlocking {
-            // "standard repositories" are mavenCentral, jcenter and google
-            val reposUrls = listOf(
-                "https://repo.maven.apache.org/maven2/",
-                "https://dl.google.com/dl/android/maven2/",
-                "https://jcenter.bintray.com/",
-                "https://plugins.gradle.org/m2/"
-            )
-            getArtifactNameToConstantMapping().map { dependencyMapping ->
-                async {
-                    getVersionCandidates(
-                        httpClient = defaultHttpClient,
-                        moduleId = ModuleId(
-                            group = dependencyMapping.group,
-                            name = dependencyMapping.artifact
-                        ),
-                        repoUrls = reposUrls,
-                        currentVersion = Version("")
-                    ).also { check(it.isNotEmpty()) }
+        val mappings = runBlocking {
+
+            getArtifactNameToConstantMapping()
+                .filter { dependencyMapping ->
+                    "${dependencyMapping.group}:${dependencyMapping.artifact}" !in validatedDependencyMapping
                 }
-            }.awaitAll().also { check(it.isNotEmpty()) }
+                .map { dependencyMapping ->
+                    async {
+                        getVersionCandidates(
+                            httpClient = defaultHttpClient,
+                            moduleId = ModuleId(
+                                group = dependencyMapping.group,
+                                name = dependencyMapping.artifact
+                            ),
+                            repoUrls = reposUrls,
+                            currentVersion = Version("")
+                        ).run {
+                            val version = lastOrNull { it.stabilityLevel == StabilityLevel.Stable } ?: lastOrNull()
+                            dependencyMapping to version
+                        }
+                    }
+                }.awaitAll()
+        }
+
+        val newValidatesMappings = mappings
+            .filter { it.second != null }
+            .joinToString(prefix = "\n", separator = "\n") { (mapping, version) ->
+                "${mapping.group}:${mapping.artifact}:${version!!.value}"
```

Comment:
Why keep the version? I think we only need to know whether the given maven coordinates are on the repo or not.

Reply:
It's not strictly necessary but it's a useful hint that we have actually parsed the pom.xml, and may help to see whether all versions are aligned - and a rule is needed - or not @louiscad

Reply:
Well, alignment of version is out of the scope of this test IMO. It's something that we manually check when we add a dependency, and something that we could automatically infer separately.

I am concerned that the tested artifacts list, if kept as is, would have outdated versions, unless we make the test check everything every time, which would incur a penalty for each run of all tests (over one minute over a 1Gbps fiber connection from my experience).

Therefore, I think it's best to not look and keep the version at all here.

Reply:
I removed the versions